dcc-portal-ui/app/scripts/app/js/app.js

```diff
@@ -488,10 +488,13 @@
       gettextCatalog.setCurrentLanguage('en_CA');
 
       HistoryManager.addToIgnoreScrollResetWhiteList(['analysis','advanced', 'compound']);
-    
-      // Add UI Router Debug if there is a fatal state change error
-      $rootScope.$on('$stateChangeError', function () {
-        console.error('State Change Error Occurred. Error occurred with arguments: ', arguments);
+      
+      $rootScope.$on('$stateChangeError', function(event, toState, toParams) {
+        $state.go('404', {page: toState.name, id: toParams.id, url: toState.url}, {location: false});
```

Comment:
Shouldn't we go to 404 when there is any $stateChangeError? i.e spelling mistake in the state, error from api or JS errors. Because we have also removed `Notify` form the code. So there won't be any user friendly action happening for any errors.